Task4/Task4-1/main.cpp

```diff
@@ -0,0 +1,379 @@
+﻿// Task4-1.
+#include <iostream>
+#include <cmath>
+#include <string> 
+#include <random>
+using namespace std;
+/**
+*\ brief находит индексы тех элементов,значения которых больше значений последующего элемента
+*\ param массив а,количество элементов массива
+*\ return индексы тех элементов,значения которых больше значений последующего элемента
+*/
+string index(int *a, int quantity);
+/**
+*\ brief находит сумму элементов,значения которых по модулю меньше 10
+*\ param массив а,количество элементов массива
+*\ return сумма элементов,значения которых по модулю меньше 10
+*/
+double summa(int *a, int quantity);
+/**
+*\ brief  Функция ввода элементов многомерного массива,как случайными числами, так и с помощью клавиатуры по желанию пользователя
+*\ param кол-во строк(n),кол-во столбцов(m),многомерный массив nˣm из n целых чисел
+*/
+void enter1(int * a, int quantity);
+/**
+*\ brief Умножает все элементы массива,кратные 3,на третий элемент массива
+*\ param кол-во строк(n),кол-во столбцов(m),многомерный массив nˣm из n целых чисел
+*/
+int krat3(int * a, int quantity);
+/**
+*\ brief  Функция ввода элементов
+*\ param сообщение о том,что нужно ввести
+*/
+int enter(const string & message);
+int main()
+{
+    const int quantity = 81;
+    int a[81];
+    enter1(a, quantity);
+cout << "the sum of elements whose values are modulo less than 10= " << summa(a, quantity) << "\n";
+cout << "indexes of those elements whose values are greater than the value of the subsequent element= "<<index(a,quantity)<<"\n";
+cout << "elements of array kratnue 3= " << krat3(a, quantity) << "\n";
+         for (int i = 0; i < quantity; i++)
+    {
+             cout << ("a[")<<i<<"]="<<a[i] << "\n";
+    }
+}
+int enter(const string & message)
+{
+    cout << message;
+    int temp;
+    cin >> temp;
+    return temp;
+}
+double summa(int * a, int quantity)
```

Comment:
если размер массива не изменяется, то нужно использовать модификатор`const`

[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.